### PR TITLE
[#112] Widened the clickable area of navLinks

### PR DESCRIPTION
**Pull request by kammanz**
What's completed:

- Expanded clickable area of nav links
- Created activeBorder styles object 
- Deleted activeStyles styles object, no longer necessary 
- Implemented withRouter and used history object to match pathnames and conditionally render border bottom

### CODE REVIEW

**Comment by kammanz** [comment]
Ready for review.

File: src/components/header/navMenu.js [hunk 1]
```diff
@@ -1,37 +1,56 @@
...
 
   return (
     <div className={styles.container}>
       <div className={styles.linkContainer}>
-        <NavLink to={playerId ? `/${urlName}/players/${playerId}` : `/${urlName}/players`} activeStyle={activeStyle}>
+        <NavLink
+          to={playerId ? `/${urlName}/players/${playerId}` : `/${urlName}/players`}
+          activeStyle={{ color: teamColor }}>
           Players
+          <div style={pathname.indexOf('/players') >= 0 ? activeBorder : null} />
```
Comment (Tumne) [inline src/components/header/navMenu.js]: Make this into a function call as it's being used elsewhere:

`  const getActiveStyle = (path) => (pathname.indexOf(path) >= 0 ? activeBorder : null);`

example:

`getActiveStyle('/players')`

File: src/components/header/navMenu.js [hunk 2]
```diff
@@ -1,37 +1,56 @@
...
+    location: { pathname },
+  },
+  selectedTeam: { teamColor, urlName },
+  playerId,
+}) => {
+  const activeBorder = {
+    position: 'absolute',
+    width: 'calc(100% - 40px)',
+    left: '20px',
+    bottom: '0',
```
Comment (Tumne) [inline src/components/header/navMenu.js]: Add these css variables as a class and no need for absolute, better like so:

```
.navLink {
  padding-top: 7px;
  border-bottom: 4px solid transparent;
}
```

example:

```
<div className={styles.navLink} style={getActiveStyle('/standings')} />
```

File: src/components/header/navMenu.module.scss [hunk 3]
```diff
@@ -12,15 +12,16 @@
...
     height: 100%;
-    padding: 0 25px;
+    padding: 0;
   }
 
   a {
     display: inline-block;
     height: calc(100% - 19px);
-    padding-top: 15px;
+    padding: 15px 20px 0;
```
Comment (Tumne) [inline src/components/header/navMenu.module.scss:22]: change to this:

```
  a {
    display: inline-block;
    height: calc(100% - 14px);
    padding: 15px 20px 0;
    color: $grey5;
    text-decoration: none;

    &:hover {
      color: $grey-three;
    }
  }
```

File: src/components/header/navMenu.js [hunk 4]
```diff
@@ -1,37 +1,56 @@
...
     borderBottom: `4px solid ${teamColor}`,
```
Comment (Tumne) [inline src/components/header/navMenu.js:19]: Now this can be change to:

```
  const activeBorder = { borderColor: teamColor };

```

**Review by Tumne — CHANGES_REQUESTED** [review]
It's too bad activeStyles can't be set in the child's components, otherwise it would've been a simple task. Regardless, great job solving the border issue.

The only thing I noticed is that the NavLink component doesn't completely reach the bottom:

Screen Shot 2020-06-28 at 8 18 24 AM

Changes required

**Comment by kammanz** [comment]
PR ready for review.

File: src/components/header/navMenu.js [hunk 5]
```diff
@@ -1,37 +1,54 @@
...
+}) => {
+  const activeBorder = {
     borderBottom: `4px solid ${teamColor}`,
   };
 
+  const getActiveStyle = (path) => (pathname.indexOf(path) >= 0 ? activeBorder : null);
+
   return (
     <div className={styles.container}>
       <div className={styles.linkContainer}>
```
Comment (Tumne) [inline src/components/header/navMenu.js]: No need for linkContainer anymore, remove those divs throughout this component and the `.linkContainer` css

File: src/components/header/navMenu.js [hunk 6]
```diff
@@ -1,37 +1,54 @@
...
+    location: { pathname },
+  },
+  selectedTeam: { teamColor, urlName },
+  playerId,
+}) => {
+  const activeBorder = {
     borderBottom: `4px solid ${teamColor}`,
   };
 
+  const getActiveStyle = (path) => (pathname.indexOf(path) >= 0 ? activeBorder : null);
```
Comment (Tumne) [inline src/components/header/navMenu.js:22]: Let's add back `activeStyle` variable and make the cursor default if navLink is selected:

```
const activeStyle = { color: teamColor, cursor: 'default' };

...
<NavLink to={playerId ? `/${urlName}/players/${playerId}` : `/${urlName}/players`} activeStyle={activeStyle}>
...
```

File: src/components/header/navMenu.js (hunk 4 above)
Comment (Tumne) [reply src/components/header/navMenu.js:19]: Still not completed

**Review by Tumne — CHANGES_REQUESTED** [review]
Couple of more changes required

File: src/components/header/navMenu.js (hunk 6 above)
Comment (Tumne) [reply src/components/header/navMenu.js:22]: I see this is an issue in PR #131, you can leave this the way it is as it's fixed in the other PR.

Comment (kammanz) [reply src/components/header/navMenu.js:18]: Nice touch

**Comment by kammanz** [comment]
Ready for review.

**Review by Tumne — APPROVED** [review]
🚀

**Pull request merged**